keras_cv/models/convmixer.py

```diff
@@ -112,7 +112,7 @@ def apply(x):
     return apply
 
 
-def patch_embed(dim, patch_size):
+def PatchEmbed(dim, patch_size):
```

Comment:
can we replace this with a class based layer component?

Reply:
Done!

Reply:
I think this will probably break our pre-trained weights for ConvMixer (we only have 1).

@IMvision12 can you test if with this change you can still load the pre-trained weights?
If not, we'll need to convert them.

Reply:
@ianstenbit I am getting this error on colab
`Exception: URL fetch failure on https://storage.googleapis.com/keras-cv/models/convmixer_512_16/imagenet/classification-v0.h5: 403 -- Forbidden`

Reply:
using given code : 
```
from keras_cv.models.convmixer import ConvMixer_512_16
model = ConvMixer_512_16(include_rescaling=False, include_top=True, classes=1000, weights='imagenet')
```

Reply:
Nope still getting same error!

Reply:
how about now?

Reply:
So, the problem is solved but there's mismatch in layers :
`ValueError: Layer count mismatch when loading weights from file. Model expected 66 layers, found 67 saved layers.`

Reply:
Yeah, that's what I expected :/

We can either port these weights manually (for both the top and no-top versions), or we can skip on this part of the change for now.

I'm fine either way. Porting the weights isn't too complex. I'm not able to do it right now but can probably get to it late this week if needed.

Reply:
@LukeWood what should we do?

Reply:
@ianstenbit I guess we can skip this part for now, we can address this part when we will add weights for all other models!!

Reply:
That sgtm -- leaving this as a method, not a layer, is fine for now.

Reply:
So, no more changes for me for the time being, right?

Reply:
Sorry I think I misunderstood you.
I don't think we should skip porting the weights. I meant let's skip making PatchEmbed a layer for now.

Currently, it's still a layer.

Reply:
Okay I will change it back to function!!

Reply:
Done!!

Reply:
sg -- thanks!